agentendpoint/retry.go

```diff
@@ -25,9 +25,19 @@ import (
 	"google.golang.org/grpc/status"
 )
 
-func retrySleep(i int, rnd *rand.Rand) time.Duration {
-	// Always increasing with some jitter, longest wait will be 5min.
-	nf := math.Min(float64(i)*float64(i)+float64(rnd.Intn(i)), 300)
+func retrySleep(i int, e int, rnd *rand.Rand) time.Duration {
+	// i=1 and e=0 => 1*1+[0,1] => 1-2s
+	// i=2 and e=0 => 2*2+[0,2] => 4-6s
+	// i=3 and e=0 => 3*3+[0,3] => 9-12s
+
+	// i=1 and e=5 => 6*1+[0,6] => 6-12s
+	// i=2 and e=5 => 7*2+[0,7] => 14-21s
+	// i=3 and e=5 => 8*3+[0,8] => 24-32s
+
+	// i=1 and e=10 => 11*1+[0,11] => 11-22s
+	// i=2 and e=10 => 12*2+[0,12] => 24-36s
+	// i=3 and e=10 => 13*3+[0,13] => 39-52s
+	nf := math.Min(float64(i+e)*float64(i)+float64(rnd.Intn(i+e)), 300)
```

Comment:
This might be easier to read as `nf := math.Min(float64((i+e)* i + rnd.Intn(i+e)), 300)`